core/include/detray/navigation/intersection/helix_plane_intersector.hpp

```diff
@@ -100,7 +106,18 @@ struct helix_intersector_impl<cartesian2D<algebra_t>, algebra_t> {
         // Build intersection struct from helix parameters
         sfi.path = s;
         sfi.local = mask.to_local_frame(trf, h.pos(s), h.dir(s));
-        sfi.status = mask.is_inside(sfi.local, mask_tolerance);
+        sfi.cos_incidence_angle =
+            vector::dot(mask.local_frame().normal(trf, sfi.local), h.dir(s));
+
+        scalar_type tol{mask_tolerance[1]};
+        if (detail::is_invalid_value(tol)) {
+            // Due to floating point errors this can be negative if cos ~ 1
+            const scalar_type sin_inc2{math::abs(
```

Comment:
Can you try to find a unified way for both ray and helix intersector?

Reply:
The point is, that the unified way does not work well. Ray and helix intersectors are different types of algorithms after all. The helix intersector is now using a tolerance that is based on the (estimate of the) approximation error of the Newton algorithm, which makes no sense to use in the ray intersector, since that one is in itself not an iterative algorithm.  

The ray intersector, on the other hand, is now capable of (but this is optional) using a dynamic tolerance to overcome a good deal of the problems of the approximation error of the ray to a curved track at long stepping distances. This in turn, is navigation related and does not occur in the helix intersection algorithm. 

Since it is navigation related, I would love to keep the mask tolerance scaling out of the ray intersector implementation entirely, but it needs the pathlength. What would make sense to do though, is split up the intersectors into the part that calculates the intersection data and the part that evaluates it and fills the resulting intersection struct. Sort of like the cylinder intersector